source/includes/api/_vocabulary.md

```diff
@@ -1,694 +1,1297 @@
+## What are resources?
 
-The Vocabulary can be described as a cluster of tags.
+Throughout this section, we'll refer multiple times to `resources`. Simply put, a `resource` is either a dataset, a
+widget, or a layer. Tags and vocabularies work in a similar fashion for each of these 3 types of RW API entity, so often
+we'll refer to resources instead, for convenience.
 
-It is currently possible to just create a Vocabulary without any tags or relationships.
-To do it, the only parameter that is required is the name of the Vocabulary.
+We assume that, at this point, you are already familiar with the concepts of [dataset](#dataset), [layer](#layer)
+or [widget](#widget) (ideally you are familiar with the 3). If that's not the case, we strongly encourage you to learn
+about those concepts first, and we also recommend you take some time to explore and experiment using
+actual [dataset](#dataset6), [layer](#layer8), or [widget](#widget9) endpoints before proceeding. Vocabularies and tags
+exist to improve and empower your use cases of those resources, so it should only be used by applications and users that
+are already comfortable with the basics, and want to take their knowledge of the RW API to the next level.
 
-| Field             | Description                                                       | Type
-| ------------------|:-----------------------------------------:                        | -----:
-| name              | The name of the Vocabulary (**it has to be unique and it represents the "primaryKey"**) | String
-| application  | The associated application for this vocabulary. Read more about the `application` field [here](/index-rw.html#applications). | String
+The behavior of vocabulary and tags endpoints aims to be, as much as possible, independent from the type of the target
+resource it references. In the detailed endpoint documentation below we'll cover the different endpoints in depth, and
+highlight the differences in behavior when handling different resource types, but you can safely assume that, for most
+of it, the behavior described for a type of resource will be the same for all 3 types.
 
-However, the most common use case relies on the relationships creation.
+### A note on resources
 
-## Relationships
+> Creating a new vocabulary/tag for a widget
 
-This section describes how to associate an existing or new Vocabulary to a resource.
+```shell
+curl -X POST https://api.resourcewatch.org/v1/dataset/AAA/widget/BBB/vocabulary/VVV \
+-H "Content-Type: application/json"  -d \
+ '{
+   "application": <application>,
+   "tags": [<tags>]
+  }'
+```
+
+In the context of the vocabulary/tag service, widgets and layers are identified not only by their own id but also by the
+id you specify as being their associated dataset. Using the side example as reference, this API request would create a
+vocabulary `VVV` for the resource widget `BBB` associated with dataset `AAA`. If you later reference the same widget id,
```

Comment:
No, this should never happen, AFAIK. Layers and widgets should only belong to a single dataset, so these types of cases should probably issue an error rather than have this unexpected behavior. But as that's not the case, I preferred to give readers a heads up. I will clarify that this is unintentional behavior

Reply:
👍